deepspeed/utils/timer.py

```diff
@@ -167,62 +171,71 @@ def __init__(
 
     def update_epoch_count(self):
         self.epoch_count += 1
-        self.local_step_count = 0
+        self.local_microstep_count = 0
 
     def _init_timer(self):
         self.initialized = True
 
     def start(self):
         self._init_timer()
         self.started = True
-        if self.total_step_count >= self.start_step:
+        if self.local_microstep_count >= self.start_step * self.gradient_accumulation_steps:
             torch.cuda.synchronize()
             self.start_time = time.time()
 
     def stop(self, report_speed=True):
```

Comment:
As suggested,
`def stop(self, global_step=False, report_speed=True)`

Reply:
Sounds good, will change.

Reply:
On the second though, don't we have a functional dependency between `global_step` and `report_speed`? The only case where the two have different values is at gradient accumulate step for non-0 rank, where we don't report value anyway. Or is it an instance of the debugging use-case you were referring previously? E.g. if we want to examine the contents of `tput_timer` for non-0 rank that is.

Reply:
Yes, reporting statistics on rank0 only is just a convention. But timer statistics is similarly collected by all ranks. So yes, this could be useful for debugging.